node/rpcstack.go

```diff
@@ -197,6 +203,20 @@ func (h *httpServer) start() error {
 }
 
 func (h *httpServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Permit dumb empty requests for remote health-checks (AWS)
+	if r.Method == http.MethodGet && r.ContentLength == 0 && r.URL.RawQuery == "" {
+		if r.URL.Path == "/readyz" {
+			if h.ready {
+				w.WriteHeader(http.StatusNoContent)
+			} else {
+				w.WriteHeader(http.StatusServiceUnavailable)
+			}
+			return
+		} else if r.URL.Path == "/livez" {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+	}
```

Comment:
This reads a bit awkward to me. I recommend the logic here focuses on the HTTP code to use and then call `w.WriteHeader` once afterwards.

```suggestion
                code := http.StatusNoContent
                if r.URL.Path == "/readyz" && !h.ready {
                        code = http.StatusServiceUnavailable
                }
                w.WriteHeader(code)		
	}
```

Reply:
If I introduce variable the number of lines stay the same and it does not become more readable.
 I cannot combine `r.URL.Path == "/readyz" && !h.ready` into one expression because return has to be executed in any case when URL is `/readyz`
```golang
	if r.Method == http.MethodGet {
		if r.URL.Path == "/readyz" {
			code := http.StatusNoContent
			if !h.ready 
				code = http.StatusServiceUnavailable
			}
			w.WriteHeader(code)
			return
		} else if r.URL.Path == "/livez" {
			w.WriteHeader(http.StatusNoContent)
			return
		}
	}
```